odl/tomo/operators/xray_trafo.py

```diff
@@ -176,57 +183,145 @@ def _call(self, inp):
+        if not isinstance(discr_range, DiscreteLp):
+            raise TypeError('discretized range {!r} is not a `DiscreteLp`'
+                            ' instance.'.format(discr_range))
+
+        if not isinstance(geometry, Geometry):
+            raise TypeError('geometry {!r} is not a `Geometry` instance.'
+                            ''.format(geometry))
+
+        if not geometry.has_motion_sampling:
+            raise ValueError('geometry {} does not have sampling grids for '
+                             'motion.'.format(geometry))
+
+        if not geometry.has_det_sampling:
+            raise ValueError('geometry {} does not have sampling grids for '
+                             'the detector.'.format(geometry))
+
+        backend = str(backend).lower()
+        if backend not in _SUPPORTED_BACKENDS:
+            raise ValueError('backend {!r} not supported.'
+                             ''.format(backend))
+
+        if backend == 'astra':
+            if isinstance(discr_range.dspace, CudaNtuples):
+                self._backend = 'astra_cuda'
+            elif isinstance(discr_range.dspace, Ntuples):
+                self._backend = 'astra_cpu'
+            else:
+                raise TypeError('discr_range.dspace {} must be a CudaNtuples '
+                                'or a Ntuples'.format(discr_range.dspace))
+        else:
+            self._backend = backend
+
+        if self.backend.startswith('astra'):
+            if not ASTRA_AVAILABLE:
+                raise ValueError('ASTRA backend not available.')
+            if not ASTRA_CUDA_AVAILABLE and self.backend == 'astra_cuda':
+                raise ValueError('ASTRA CUDA backend not available.')
+            if discr_range.dspace.dtype not in (np.float32, np.complex64):
+                raise ValueError('ASTRA support is limited to `float32` for '
+                                 'real and `complex64` for complex data.')
+            if not np.allclose(discr_range.grid.stride[1:],
+                               discr_range.grid.stride[:-1]):
+                raise ValueError('ASTRA does not support different voxel '
+                                 'sizes per axis (got {}).'
+                                 ''.format(discr_range.grid.stride))
+
+        self._geometry = geometry
+        self.kwargs = kwargs
+
+        # Create a discretized space (operator domain) with the same data space
+        # type as the range.
+        domain_uspace = FunctionSpace(geometry.params)
+
+        # Approximate cell volume
+        extent = float(geometry.grid.extent().prod())
+        size = float(geometry.grid.size)
+        weight = extent / size
+
+        domain_dspace = discr_range.dspace_type(
+            geometry.grid.size, weight=weight, dtype=discr_range.dspace.dtype)
 
-    def _call(self, inp):
-        """Call the adjoint transform on an input, producing a new vector.
+        domain_interp = kwargs.pop('interp', 'nearest')
```

Comment:
`pop` removes it from the dict, since you save kwargs you lose this option. Use `get` instead.